src/luarocks/cmd/new_version.lua

```diff
@@ -201,6 +202,10 @@ function new_version.command(args)
    local out_name = out_rs.package:lower()
    out_rs.version = new_rockver.."-"..new_rev
 
+   if not args.new_url and args.keep_url then
+      args.new_url = out_rs.source.url
```

Comment:
I don't think we modify `args` anywhere; consider factoring out into a new local `new_url`.

Reply:
you do at line 179
```
if args.tag and not args.new_version then
      args.new_version = args.tag:gsub("^v", "")
end
```